### PR TITLE
Fix rpm2archive behavior with multiple arguments

### PR DESCRIPTION
**Pull request by frozencemetery**
If multiple arguments are passed to rpm2archive, convert them all to
tgz.  (Previous behavior was to convert only the first one and
silently ignore the rest.)

Signed-off-by: Robbie Harwood <rharwood@redhat.com>

### CODE REVIEW

**Comment by ignatenkobrain** [comment]
It seems that your patch is broken:

```
./rpmgeneral.at:318:

runroot rpm2archive - < "${RPMTEST}"/data/RPMS/hello-2.0-1.x86_64.rpm | tar tz

runroot rpm2archive - < "${RPMTEST}"/data/SRPMS/hello-1.0-1.src.rpm | tar tz



--- /dev/null	2019-04-24 19:34:18.132790957 +0000

+++ /opt/rpm/tests/rpmtests.dir/at-groups/6/stderr	2019-04-24 19:34:23.620810286 +0000

@@ -0,0 +1,2 @@

+rpm2archive: --define: No such file or directory

+rpm2archive: --define: No such file or directory

6. rpmgeneral.at:316: 6. rpm2archive (rpmgeneral.at:316): FAILED (rpmgeneral.at:318)
```

File: rpm2archive.c [hunk 1]
```diff
@@ -187,34 +187,34 @@ static int process_package(rpmts ts, char * filename)
...
-
-    rc = process_package(ts, filename);
+    if (argc > 1 && (rstreq(argv[1], "-h") || rstreq(argv[1], "--help"))) {
+	fprintf(stderr, "Usage: %s [file.rpm ...]\n", argv[0]);
+	exit(EXIT_FAILURE);
+    }
 
-    ts = rpmtsFree(ts);
+    if (argc == 1)
+	argv[argc++] = "-";
```
Comment (ffesti) [inline rpm2archive.c]: This doesn't look right. You can't just write beyond the bounds of the argv array. You need to allocate a new - bigger - one.

Comment (frozencemetery) [reply rpm2archive.c]: Per C standard, argv[argc] is always defined to be a NULL pointer.  (Section 5.1.2.2.1, paragraph 2, in ISO/IEC 9899:1999).

Comment (frozencemetery) [reply rpm2archive.c]: Now you may not *like* that I'm doing this, which is a perfectly valid complaint and I'm happy to change it :)

**Comment by frozencemetery** [comment]
Problem was due to a CI bug, which I've included a fix for.

File: rpm2archive.c [hunk 2]
```diff
@@ -187,34 +187,34 @@ static int process_package(rpmts ts, char * filename)
...
+    for (i = 1; i < argc; i++) {
+	rpmts ts = rpmtsCreate();
+	rpmVSFlags vsflags = 0;
+
+	/* XXX retain the ageless behavior of rpm2cpio */
+	vsflags |= RPMVSF_MASK_NODIGESTS;
+	vsflags |= RPMVSF_MASK_NOSIGNATURES;
+	vsflags |= RPMVSF_NOHDRCHK;
+	(void) rpmtsSetVSFlags(ts, vsflags);
+
```
Comment (pmatilai) [inline rpm2archive.c]: All this transaction fiddling should be done exactly once, outside the loop.

File: rpm2archive.c (hunk 1 above)
Comment (pmatilai) [reply rpm2archive.c]: Yeah, we don't like that. Please change :)

**Comment by ffesti** [comment]
Pushed with minor tweaks. Thanks for the patch and the additional effort of fixing the test suite!

**Pull request closed without merging**